spring-integration-core/src/main/java/org/springframework/integration/support/MutableMessageHeaders.java

```diff
@@ -38,13 +40,7 @@
 	private static final long serialVersionUID = 3084692953798643018L;
 
 	public MutableMessageHeaders(@Nullable Map<String, Object> headers) {
-		this(headers,
-				(headers != null ?
-						(UUID) headers.get(MessageHeaders.ID)
-						: null),
-				(headers != null ?
-						(Long) headers.get(MessageHeaders.TIMESTAMP)
-						: null));
+		super(headers, MutableMessageHeaders.extractId(headers), MutableMessageHeaders.extractTimestamp(headers));
```

Comment:
We don't need class name if `static` method is here as well.

Also your name for the modified test class.

I'll fix on merge.

Thank you!